common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -266,28 +273,21 @@ def test_email_field(self):
             assert_after_reload=False
         )
 
-        self.verify_events_of_type(
-            username,
-            self.CHANGE_INITIATED_EVENT_NAME,
+        actual_events = self.wait_for_events(
+            event_filter=self.settings_change_initiated_event_filter, number_of_matches=2)
```

Comment:
What happens if the code accidentally emits three events? I think that this code will return once the first two are captured and the test will pass if they are as expected. I think my old tests had the same behavior so I'm fine with saying that this is just how it works, but is it possible to do something better? Is there a way to know that we've reached a steady state where no more events could be emitted? I suppose that's difficult for client-side events.